wordpress-mobile/WordPress-Android
# Avoid analytics call in story composer when analytics wrapper is null

Fixes a crash that occurs if the editor activity is destroyed by the system when composing a Story Block (causes the app to try to track analytics using a null analytics wrapper). This shows up in Sentry [here](https://sentry.io/share/issue/5611e3cfe89446a8a6bfd4b4c70df1a1/). 

This causes the editor Activity to crash, which drops the user back out to the main app (so it doesn't really look like a crash to the user). Reloading the post in the editor would reveal that the story block had not been updated with the selected image/story.

This crash was occurring because when the StoryComposerActivity gets recreated, we were attempting to directly deserialize the `PostEditorAnalyticsSession` instance from the bundle. Instead, we need to use the `PostEditorAnalyticsSession::fromBundle` helper method that prevents the [`transient AnalyticsTrackerWrapper` field](https://github.com/wordpress-mobile/WordPress-Android/blob/0fdbd0adfa5c99e770abfb14edfbf06e84614e3f/WordPress/src/main/java/org/wordpress/android/ui/posts/PostEditorAnalyticsSession.java#L37) (which was added [late last year](https://github.com/wordpress-mobile/WordPress-Android/pull/15465/files#diff-471b17a541b67be8b228ee74e749614c81c28bed72dad82663e442b19d876c5eR37)) from [being null](https://github.com/wordpress-mobile/WordPress-Android/blob/0fdbd0adfa5c99e770abfb14edfbf06e84614e3f/WordPress/src/main/java/org/wordpress/android/ui/posts/PostEditorAnalyticsSession.java#L70).

I did some searching, and I think this is the only place where we were hitting this problem.

To test:
1. Turn on "Don't keep activities" in developer settings
2. Open the Gutenberg Editor
3. Add a story block
4. Tap add media on the story block
5. In the select image screen, select an image and tap the confirmation checkmark
6. In the story composer screen, just tap the confirmation checkmark
7. Observe the editor re-loads with the selected image displayed in the Story block, and that in the logs there is an error message "Unable to track session end, mAnalyticsTrackerWrapper is null".

## Regression Notes
1. Potential unintended areas of impact
None that I can think of


2. What I did to test those areas of impact (or what existing automated tests I relied on)
N/A

3. What automated tests I added (or what prevented me from doing so)
N/A

PR submission checklist:
